src/horus_api.h

```diff
@@ -35,20 +35,75 @@
 #include <stdint.h>
 #include "modem_stats.h"
       
-#define HORUS_MODE_BINARY            0
-#define HORUS_MODE_RTTY              1
+/* Horus API Modes */
+#define HORUS_MODE_BINARY            0  // Legacy binary mode
+#define HORUS_MODE_BINARY_256BIT     1  // New 256-bit LDPC-encoded mode
+#define HORUS_MODE_BINARY_128BIT     2  // New 128-bit LDPC-encoded mode
+#define HORUS_MODE_RTTY              99 // RTTY Decoding
+
+
+// Settings for Legacy Horus Binary Mode (Golay Encoding)
+#define HORUS_BINARY_NUM_BITS                   360   /* fixed number of coded bits */
+#define HORUS_BINARY_NUM_PAYLOAD_BYTES          22    /* fixed number of bytes in binary payload     */
+#define HORUS_BINARY_DEFAULT_BAUD               100   /* Default baud rate of 100 baud */
+#define HORUS_BINARY_DEFAULT_TONE_SPACING       270   /* Default tone spacing of 270 Hz */
+
+// Settings for Horus Binary 256-bit mode (LDPC Encoding, r=1/3)
+#define HORUS_BINARY_256BIT_NUM_BITS            768
+#define HORUS_BINARY_256BIT_NUM_PAYLOAD_BYTES   32
+#define HORUS_BINARY_256BIT_DEFAULT_BAUD        100   /* Default baud rate of 100 baud */
+#define HORUS_BINARY_256BIT_DEFAULT_TONE_SPACING 270   /* Default tone spacing of 270 Hz */
+
+// Settings for Horus Binary 128-bit mode (LDPC Encoding, r=1/3)
+#define HORUS_BINARY_128BIT_NUM_BITS            384
+#define HORUS_BINARY_128BIT_NUM_PAYLOAD_BYTES   16
+#define HORUS_BINARY_128BIT_DEFAULT_BAUD        50   /* Default baud rate of 100 baud */
+#define HORUS_BINARY_128BIT_DEFAULT_TONE_SPACING 270   /* Default tone spacing of 270 Hz */
+
+// Settings for RTTY Decoders
+#define HORUS_RTTY_NUM_BITS                     1000 /* Maximum 1000 bits in a RTTY frame (100 characters) */
+#define HORUS_RTTY_DEFAULT_BAUD                 100 /* Default baud rate of 100 baud */
 
 struct horus;
 struct MODEM_STATS;
 
-struct horus *horus_open  (int mode, int Rs);
+/*
+ * Create an Horus Demod config/state struct using default mode parameters.
+ * 
+ * int mode - Horus Mode Type (refer list above)
+ */
+struct horus *horus_open  (int mode);
+
```

Comment:
Does this behave as per V1 (peak estimator, so assumptions about Tx shift?)

Reply:
Nope. 
I've made the default settings for the legacy horus binary mode take advantage of the new mask estimator with the known shift (270 Hz) of the only transmitter of this mode that we have in service - the RS41. 
The defaults for the new modes are subject to change.

Reply:
This and the HW lock-in doesn't work for me - when convenient lets have a phone call to discuss this and several other aspects raised in my review.